app/util/Tracing.js

```diff
@@ -0,0 +1,331 @@
+/**
+ * Util class for tracing related functions.
+ *
+ * @class CpsiMapview.util.Tracing
+ */
+Ext.define('CpsiMapview.util.Tracing', {
+
+    statics: {
+
+        /**
+         * Compute the modulo for negative values.
+         *
+         * Example: staticMe.computeModulo(-1, 4) returns 3
+         *
+         * Taken from the official OpenLayers example:
+         * https://openlayers.org/en/latest/examples/tracing.html
+         *
+         * @param {Number} a The first number
+         * @param {Number} b The second number
+         *
+         * @returns {Number} The modulo
+         */
+        computeModulo: function (a, b) {
+            return ((a % b) + b) % b;
+        },
+
+        /**
+         * Checks if a LineString is not empty.
+         *
+         * @param {ol.Feature} feature The feature to check
+         * @returns {Boolean} If LineString is not empty
+         */
+        lineStringNotEmpty: function (feature) {
+            if (!feature) {
+                return;
+            }
+            var geom = feature.getGeometry();
+            if (!geom) {
+                return;
+            }
+            var coords = geom.getCoordinates();
+            if (!coords) {
+                return;
+            }
+            return coords.length > 0;
+        },
+
+        /**
+         * Finds the index of a coordinate in a coordinate array
+         * by only comparing its value.
+         *
+         * @param {ol.coordinate.Coordinate[]} coordinateArray The coordinate array
+         * @param {ol.coordinate.Coordinate} coordToFind The coordinate to find
+         * @returns {Number} The index of the coordinate, or -1 if not found
+         */
+        getCoordIndex: function (coordinateArray, coordToFind) {
+            var found = Ext.Array.findBy(coordinateArray, function (c) {
+                return Ext.Array.equals(c, coordToFind);
+            });
+            return coordinateArray.indexOf(found);
```

Comment:
If you use `Ext.Array.reduce` instead of `Ext.Array.findBy` you can omit the second loop.

Reply:
I moved this function to the follow-up PR #578  and therefore moved this note there